multidimensional-pod-autoscaler/README.md

```diff
@@ -0,0 +1,134 @@
+
+Same as VPA, MPA is configured with a [Custom Resource Definition object](https://kubernetes.io/docs/concepts/api-extension/custom-resources/)
+called [MultidimPodAutoscaler](https://github.com/IIDA-Institute/autoscaler/blob/mpa-dev/multidimensional-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1alpha1/types.go).
+It allows to specify which pods should be vertically and horizontally autoscaled as well as if/how the resource recommendations are applied.
+
+To enable multi-dimensional pod autoscaling on your cluster please follow the installation procedure described below.
+
+## Prerequisites
+
+* `kubectl` should be connected to the cluster you want to install MPA.
+* The metrics server must be deployed in your cluster. Read more about [Metrics Server](https://github.com/kubernetes-incubator/metrics-server).
+* If you are using a GKE Kubernetes cluster, you will need to grant your current Google
+  identity `cluster-admin` role. Otherwise, you won't be authorized to grant extra
+  privileges to the MPA system components.
+  ```console
+  $ gcloud info | grep Account    # get current google identity
+  Account: [myname@example.org]
+
+  $ kubectl create clusterrolebinding myname-cluster-admin-binding --clusterrole=cluster-admin --user=myname@example.org
+  Clusterrolebinding "myname-cluster-admin-binding" created
+  ```
+* You should make sure your API server supports Mutating Webhooks.
+  Its `--admission-control` flag should have `MutatingAdmissionWebhook` as one of
+  the values on the list and its `--runtime-config` flag should include
+  `admissionregistration.k8s.io/v1beta1=true`.
+  To change those flags, ssh to your API Server instance, edit
+  `/etc/kubernetes/manifests/kube-apiserver.manifest` and restart kubelet to pick
+  up the changes: ```sudo systemctl restart kubelet.service```
+* Please upgrade `openssl` to version 1.1.1 or higher (needs to support `-addext` option).
+
+## Installation
+
+To install MPA, please download the source code of MPA
+and run the following command inside the `multidimensional-pod-autoscaler` directory:
+
+```
+./deploy/mpa-up.sh
+```
+
+Note: the script currently reads environment variables: `$REGISTRY` and `$TAG`.
+Make sure you leave them unset unless you want to use a non-default version of MPA.
+
+The script issues multiple `kubectl` commands to the
+cluster that insert the configuration and start all needed pods
+in the `kube-system` namespace. It also generates
+and uploads a secret (a CA cert) used by MPA Admission Controller when communicating
+with the API server.
+
+## Tearing Down
+
+To remove MPA installation:
+
+```
+./deploy/mpa-down.sh
+```
+
+## Quick Start
+
+After [installation](#installation) the system is ready to recommend and set
+resource requests for your pods.
+In order to use it, you need to insert a *Multidimensional Pod Autoscaler* resource for
```

Comment:
Is "*Multidimensional Pod Autoscaler*" here the same as MultidimPodAutoscaler in line 10? If so please use the same name